### ISSUE TITLE
[feat] Add app navbar

### ISSUE THREAD

**Issue by adrianvrj** [web3wagers/gostarkme#85]
### This issue will be part of ODHack8.0, please apply via Onlydust app
- When going to the funding application, there will be a common navbar for all routes, add the necessary code to create the component that will be used as the navbar. The navbar should look like this:
![image](https://github.com/user-attachments/assets/4f3388f1-5263-43ca-aa72-7d4fd00e35f8)

    Please do not add any functionality since this will be part of another story once the full design of the web application is finished.
    
- Knowledge in Nextjs and typescript are recommended.

- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue.

**Comment by raizo07** [comment]
Hello I want to work on this

**Comment by Jemiiah** [comment]
I would love to work on this @adrianvrj

**Comment by sanieni6** [comment]
Hey @adrianvrj. i'm gonna be glad to work on this😄
I work as a React developer and I use Next.js in my daily basis
My approach:
I’ll start by creating a reusable Navbar component using Next.js and TypeScript, following the design provided. I will ensure that the component structure is modular and adheres to best practices, making it easy to integrate functionality in the future.

**Comment by coxmars** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
I'm a software developer with 2+ years of experience and I have worked with backend technologies such as Cairo, Java, C# and frontend with React, NextJS, JS/TS. Also, I'm a member of the Dojo Coding community here in CR and I have made contributions in Cairo VM Go, Cairo VM TS, Cairo Lints etc.

### How I plan on tackling this issue
1- I’ll review the contributor guide and any existing components to ensure the navbar follows the project’s structure and styling conventions.
2- I’ll create a reusable Navbar component using Next.js and TypeScript, ensuring it matches the design provided in the image.
3- I’ll make sure the component is responsive and can be easily integrated into all routes, without adding any functionality as specified.
4- Finally, I’ll test the component across different routes to verify that it works as intended visually.

**Comment by sanieni6** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hi! My name is Luis Sanchez, and I’m a Frontend Developer with 3 years of experience, specializing in TypeScript, React, and Next.js. I am a Dojo coding member and i would be happy to contribute to this project.
I work as a React developer and I use Next.js in my daily basis

### How I plan on tackling this issue
I’ll start by creating a reusable Navbar component using Next.js and TypeScript, following the design provided. I will ensure that the component structure is modular and adheres to best practices, making it easy to integrate functionality in the future.

**Comment by martinvibes** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hi I would love to work on this issue

**Comment by sanieni6** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Note: TRYING AGAIN BECAUSE OF A ONLYDUST ISSUE

Hi! My name is Luis Sanchez, and I’m a Frontend Developer with 3 years of experience, specializing in TypeScript, React, and Next.js. I am a Dojo coding member and i would be happy to contribute to this project.
I work as a React developer and I use Next.js in my daily basis

### How I plan on tackling this issue
I’ll start by creating a reusable Navbar component using Next.js and TypeScript, following the design provided. I will ensure that the component structure is modular and adheres to best practices, making it easy to integrate functionality in the future.

**Comment by raizo07** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hello I want to work on this


### How I plan on tackling this issue
ETA: One day max.

**Comment by CollinsC1O** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hello I'm a front-end and a blockchain developer and i will love to work on this

**Comment by Benjtalkshow** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hi @adrianvrj,

I’m Benjamin, a JavaScript developer with 4 years of experience and over 18 contributions on OnlyDust | https://app.onlydust.com/u/Benjtalkshow. I’m really passionate about creating responsive and user-friendly interfaces, especially with tools like Next.js and Svelte. Having worked on the transaction details and blocks details pages for Opscan, I’m excited about the chance to help out with the navbar for Go Stark Me Project.

### How I plan on tackling this issue
For this task, I’ll be putting together a common navbar that will be used across all routes in our application. I’ll take a good look at the design you provided in the image to ensure it looks just right. My goal is to make it visually appealing and seamless with the rest of the site.
I’ll also focus solely on the design aspect for now, so I won’t be adding any functionality just yet. That’ll come later, once we have the full design laid out.

I’ll make sure to stick to the project’s style guide so that everything stays consistent. Using Next.js and TypeScript will help me keep the code clean and easy to maintain, which I know is important for future updates.

**Comment by GoSTEAN** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
I have experience in frontend development using Next.js and TypeScript, along with a solid understanding of component-based architecture. I've built reusable components and styled them according to design specifications in various web applications. This experience will allow me to create a scalable and efficient navbar for your project while following best practices.




### How I plan on tackling this issue
I would begin by reviewing the contributor's guide to ensure alignment with project standards. Next, I would create a reusable and responsive Navbar component in Next.js using TypeScript. I would follow the provided design closely to match the visual layout and ensure that it can be easily extended in future tasks. Functionality will be left out, as per the requirement, and I will focus purely on structure and design.

**Comment by wheval** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
GM, I'm wheval a frontend developer with multiple experience converting figma designs into pixel perfect responsive websites.  Recently, i worked on a project where i built a dashboard UI for a web3 startup from their figma design using react and tailwindcss. I have also participated in a X/Twittter challenge where i built and interactive dashboard using Nextjs and typescript. Coupled with my other experiences workig with nextxjs' app router, i can deliver this issue perfectly well and also with a good timing

### How I plan on tackling this issue
Following the contributors guide, firstly i would fork the repo and create a new branch like git branch -b feat/navbar from the dev branch. Then, in the component folder in the gostarkme-web directory, i would create a new folder named "navbar" inside the navbar folder, i would create a new file named Navbar.tsx which would be the navbar component. From the figma file, i would get the necessary images icons needed and put them in the "public/images" and "public/icons" directories respectively.
I estimate this will take to complete.
For any enquiries, i would reach out to ADRIAN on the Discord. Thank you.

**Comment by josephchimebuka** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hello, I am Joseph I am a frontend developer and blockchain developer and I am also an active contributor here on only dust here is my profile https://app.onlydust.com/u/josephchimebuka. This is my first time to contribute to this repo ill appreciate the opportunity to contribute. this is a couple of frontend and blockchain projects i have built https://metacrypt.vercel.app/ https://mattedsgn.com/

### How I plan on tackling this issue
I will follow the decription criteria and ensure that the app navbar is added. I will conclude it in 2-3 days time

**Comment by MariangelaNM** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hello, I'm Mariangela. I'm a software developer specializing in front-end development.

### How I plan on tackling this issue
My first step will be to thoroughly review the contributor’s guide to ensure I adhere to the project’s coding standards, design principles, and best practices. Once aligned, I will proceed to develop a reusable, responsive Navbar component in Next.js using TypeScript. This component will be crafted to ensure it integrates seamlessly with the rest of the application while maintaining a high degree of modularity for future scalability. I'll pay close attention to responsiveness and adaptability across different screen sizes, following the provided design meticulously to capture the exact visual appearance and user experience.

In addition, I will document the component's structure and styling choices, making it easier for other developers to understand and extend its functionality in subsequent iterations. Since the current focus is solely on the layout and design, I will omit any interactive features or dynamic functionality, as per the project’s requirements. My goal is to deliver a well-structured, visually accurate component that can be effortlessly enhanced in the future.

**Comment by melnikga** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Can I take this? I’ve participated in almost every ODHack. I have extensive experience in frontend development. I can handle this task.





### How I plan on tackling this issue
In all the projects I worked on during ODHacks, I used Next.js, so implementing this page will be easy for me.

My profile on OD: https://app.onlydust.com/u/melnikga

**Comment by Solomonsolomonsolomon** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Fullstack typescript developer

### How I plan on tackling this issue
Easy issue,I will create a Navbar component and style it accordingly

**Comment by abdegenius** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
I am a skilled fullstack developer with a wealth of experience in both frontend and backend JavaScript as well as my knowledge of Rustlang, With a solid foundation in testing,I have mastered frameworks like Mocha, Chai, and Jest to ensure robust, reliable code in every project. In addition to traditional web development, I have a strong background in blockchain technology, with hands-on experience in Solidity for Ethereum-based smart contracts. Moreover, I have delved into the emerging world of StarkNet, working with Cairo, a cutting-edge language tailored for STARKs, making them a versatile developer capable of bridging both traditional and decentralized web applications.


### How I plan on tackling this issue
I will complete the replication of the provided figma design into a nextjs page using typescript, making sure to use component based design and write well documented, commented and bug free code.

**Comment by fabrobles92** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
I am a full stack engineer with experience in Next Js, Vue and plain react, also I have plenty of contribution to web3 projects where that experience can help me deliver a qulaity solution

### How I plan on tackling this issue
I would add the necessary tailwind classes to create the component and then create a pluggable react component <Navbar /> and then add it in page.tsx so it always appear in different screens

**Comment by chibokaxavier** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
Hi, I'm a Full-Stack Developer with  over 4 years of experience in building web applications. I specialize in Next.js, React.js, Node.js, Tailwind CSS, and TypeScript, and I've spent the past few years developing everything from sleek, responsive frontends to powerful, efficient backend systems.

What sets me apart is my passion for delivering clean, well-structured code and user-friendly interfaces. I love turning complex problems into elegant, practical solutions. Whether it's crafting dynamic, high-performance applications or integrating robust APIs, I bring a balance of technical know-how and creativity to every project.

### How I plan on tackling this issue
Step 1: Create the Navbar Component
I will create a new component for the navbar:

I will create a file called components/Navbar.tsx. This will house the navbar component that will be used across all routes.

Step 2: Define the Structure
I will outline the structure of the navbar according to the design:

I will analyze the provided image to determine the layout, such as the placement of the logo, navigation links, and any icons.
I will create a basic HTML structure within the component, using semantic elements like <nav>, <ul>, and <li> to represent the links.

Step 3: Style the Navbar
I will ensure the navbar matches the visual design:

I will apply CSS classes, likely using Tailwind CSS (or my preferred styling method), to achieve the look and feel depicted in the image.
I will consider aspects like spacing, colors, hover effects, and font styles to ensure consistency with the overall design.

Step 4: Add Placeholder Links
While I will not implement any functionality, I will add placeholder links:

I will create <Link> components for navigation, providing dummy paths that reflect the intended routes (e.g., Home, My Funds, etc.).
I will ensure these links are visually represented as buttons or menu items in the navbar.

Step 5: Responsive Design
I will make sure the navbar is responsive:

I will implement responsive design practices, ensuring the navbar adjusts correctly on different screen sizes.
I might include a hamburger menu icon for mobile views, though without functionality for now.

Step 6: Follow Best Practices
I will ensure the code follows best practices:

I will use TypeScript to type any props or state if necessary, although it may be minimal for this component.
I will adhere to any coding standards outlined in the contributors' guide to maintain consistency in the project.

Step 7: Embed in Layout File
If there is a layout file in Next.js, I will embed the navbar there:

I will look for an existing layout file, typically located in pages/_app.tsx or a similar structure.
I will include the <Navbar /> component within the layout file so that it appears consistently across all pages.

Step 8: Skip Functionality for Now
I will refrain from adding any functionality:

The focus will be solely on the design and structure of the navbar, deferring any interactive elements or routing logic to a later development stage.

**Comment by Rayan9064** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/go-stark-me).

### My background and how it can be leveraged
I am a Full Stack Developer and Cairo Developer. I have good experience building web application. I am new to Open Source and want to begin with this project. I am good in React, Next.js, Typescript, Redux and Context APIs. I can leverage the same to make this a better game for the users. 

### How I plan on tackling this issue
1) Checking the structure of the project and understanding the codebase. 
2) Knowing the tech stack required
3) Finding the area of the problem. 
3) Observe the Figma design. 
4) Implementing into code. 
5) Testing the code by inspecting in different screen size. 
6) Modify the code when not working as expected. 
7) After successful review, create a pull request.
